src/backends/torch/torchgraphbackend.h

```diff
@@ -0,0 +1,173 @@
+  private:
+	std::string _s;
+  };
+
+  /**
+   * \brief torch module that includes a basegraph,
+   * implements forward based on basegraph
+   * redefines to() and parameters()
+   * allocates torch modules based on embedded basegraph
+   * should not be used as is
+   * it only add torch::module traits ie to(), forward() parameters()...  to a base graph
+   */
+  class TorchGraphBackend : public virtual BaseGraph, public torch::nn::Module
+  {
+  public:
+
+	/**
+	 * \brief
+	 * empty constructor : do nothing this class is not meant to be constructed alone
+	 *
+	 */
+	TorchGraphBackend() {}
+
+	/**
+	 * \brief torch::module forward :  inference of the NN
+	 * @param x input tensor
+	 * @return output tensor
+	 */
+	torch::Tensor forward(torch::Tensor x);
+
+	/**
+	 * \brief allocates torch modules base on real input dimension
+	 * @param inputdim input tensor dimension, in int64_t
+	 */
+	void finalize(std::vector<int64_t> inputdim);
+
+	/**
+	 * \brief allocates torch modules base on real input dimension
+	 * @param inputdim input tensor dimension, in int
+	 */
+	void finalize(std::vector<int> inputdim);
+
+	/**
+	 * \brief allocates torch modules base on real input dimension
+	 * @param inputdim input tensor dimension, in torch :: IntArrayRef
+	 */
+	void finalize(at::IntArrayRef inputdim);
+
+	/**
+	 * \brief allocates torch modules base on current
+	 * hides basegraph::finalize() on purpose
+	 * generally should not be used except for very special cases
+	 */
+	void finalize();
+
+	/**
+	 * \brief see torch::module::to
+	 * @param device cpu / gpu
+	 * @param non_blocking
+	 */
+	virtual void to(torch::Device device, bool non_blocking = false);
+
+	/**
+	 * \brief see torch::module::to
+	 * @param dtype : torch::kFloat32 or torch::kFloat64
+	 * @param non_blocking
+	 */
+	virtual void to(torch::Dtype dtype, bool non_blocking = false);
+
+	/**
+	 * \brief see torch::module::to
+	 * @param device cpu / gpu
+	 * @param dtype : torch::kFloat32 or torch::kFloat64
+	 * @param non_blocking
+	 */
+	virtual void to(torch::Device device, torch::Dtype dtype, bool non_blocking = false);
+
+	/**
+	 * see torch::modules::parameters()
+	 * override in order to warn if reallocation is needed while parameters are used
+	 * @param recurse
+	 *
+	 * @return
+	 */std::vector<torch::Tensor> parameters(bool recurse = true);
+
+	/**
+	 * set lstm continuation : if true then previous hidden state of lstm is used for upcoming forward()
+	 * @param lc
+	 */
+	void lstm_continues(bool lc) {_lstm_continuation = lc;}
+
+
+	/**
+	 * informs torchgraphbackend that parameters are no more used,
+	 * ie reallocation can be done w/o warning
+	 */
+	void parameters_release() {_parameters_used = false;}
+
+  protected:
+
+	/**
+	 * internal torch module allocation, called whithin (finalize)
+	 * @param force
+	 */
+	void allocate_modules(bool force = false);
+
+	std::map<std::string,torch::nn::AnyModule> _modules; /**< torch modules, per name/id */
```

Comment:
Is it possible use `std::unordered_map` instead ?

Reply:
done